### DISCUSSION TITLE
Elastic Agent Authentication

### DISCUSSION THREAD

**Discussion by GeorgeWere** [Security-Onion-Solutions/securityonion#10705] [2.4] [answered]
Hi SO Team,

First and foremost, I wanted to express my appreciation for the outstanding work you have done with the release of Security Onion 2.4. The concept of single agent deployment is truly commendable and I must say that I am thoroughly enjoying its benefits.

However, I have come across a concern regarding the authentication process. I have noted that the authentication token generated during installation is unique and cannot be changed afterwards. This becomes problematic in scenarios where the manager experiences downtime or requires a fresh installation, as it would necessitate reauthenticating all the agents using new tokens generated by the new installation. I was wondering if there is a workaround for this issue?

**Comment by TOoSmOotH** [comment, marked as answer]
This is a limitation of Elastic Agent itself as it is tied to the specific Elastic cluster id. In theory, as long as you maintain that clusterid and original key you should be ok to restore your manager back to an existing cluster with the manager role. If it is a standalone and it is rebuilt then you would need to re-install the agents.  This is something we will need to test more and see what the best option is when running the nightly backup to try and make it easier to recover.

**Reply by GeorgeWere** [reply]
Thank you @TOoSmOotH . Can't wait to see what you guys come up with :)
